lib/logstash/filters/lookup.rb

```diff
@@ -0,0 +1,183 @@
+#
+# The map entries can be specified with a Web service who your request produces a YML, CSV or JSON response.
+#
+# Operationally, if the event field specified in the `field` configuration
+# matches the EXACT contents of a map entry key, the field's value will be substituted
+# with the matched key's value from the map.
+#
+# By default, the lookup filter will replace the contents of the
+# maching event field (in-place). However, by using the `destination`
+# configuration item, you may also specify a target event field to
+# populate with the new mapd value.
+
+class LogStash::Filters::LookUp < LogStash::Filters::Base
+  config_name "lookup"
+  # The name of the logstash event field containing the value to be compared for a
+  # match by the map filter (e.g. `message`, `host`, `response_code`). 
+  # 
+  # If this field is an array, only the first value will be used.
+  config :field, :validate => :string, :required => true
+
+  # If the destination (or target) field already exists, this configuration item specifies
+  # whether the filter should skip mapping (default) or overwrite the target field
+  # value with the new mapping value.
+  config :override, :validate => :boolean, :default => false
+
+  # The full URI path of a Web service who generates an JSON, yml or CSV format response.
+  # requires to append as suffix the format type. Ex: http://localhost:8080/geoPoints?type=json
+  # http://localhost:8080/geoPoints/json
+  # http://localhost:8080/geoPoints/csv
+  # If no suffix matches, defaults to YAML
+  config :map_url, :validate => :string, :required => true
+
+  # When using a map file or url, this setting will indicate how frequently
+  # (in seconds) logstash will check the YAML file or url for updates.
+  config :refresh_interval, :validate => :number, :default => 300
+
+  # The destination field you wish to populate with the mapd code. The default
+  # is a field named `mapping`. Set this to the same value as source if you want
+  # to do a substitution, in this case filter will allways succeed. This will clobber
+  # the old value of the source field! 
+  config :destination, :validate => :string, :default => "mapping"
+
+  # In case no mapping occurs in the event (no matches), this will add a default
+  # mapping string, which will always populate `field`, if the match failed.
+  #
+  # For example, if we have configured `fallback => "no match"`, using this map:
+  # [source,ruby]
+  #     foo: bar
+  #
+  # Then, if logstash received an event with the field `foo` set to `bar`, the destination
+  # field would be set to `bar`. However, if logstash received an event with `foo` set to `nope`,
+  # then the destination field would still be populated, but with the value of `no match`.
+  # This configuration can be dynamic and include parts of the event using the `%{field}` syntax.
+  config :fallback, :validate => :string
```

Comment:
Thoughts on calling this `default_values`? Once `fields` take an array we can then make this an array too, and force the `len(fields) == len(default_values)`. This way, we can not worry about positions, and do a lookup based on position in `fields` to get its default value.